### PR TITLE
Fix performance bucket corner case

### PR DESCRIPTION
**Pull request by blindpirate**
Previously, because flakiness detection repeats the same test for 2 times,
2 same scenarios might be split into the same bucket, triggering an IllegalArgumentException.
This PR handles this case specially.

### CODE REVIEW

File: .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt [hunk 1]
```diff
@@ -126,14 +125,24 @@ fun splitBucketsByScenarios(scenarios: List<PerformanceScenario>, testProjectToS
...
+        ::mergeSmallTestProjectsIntoOneBucket,
         numberOfBuckets,
         MAX_TEST_PROJECTS_PER_BUCKET,
         { numEmptyBuckets -> (0 until numEmptyBuckets).map { EmptyTestProjectBucket(it) }.toList() },
         { tests1, tests2 -> tests1 != tests2 }
     )
 }
 
+fun mergeSmallTestProjectsIntoOneBucket(smallProjectDuration: List<TestProjectDuration>): PerformanceTestBucket {
+    // Because of flakiness detection, these "smallProjectDuration" actually can contain same testProject
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:137]: But that is not good, right? If that happens, we will only run re-run the flaky test once and not multiple times.

Comment (blindpirate) [reply .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:137]: The `IllegalArgumentException` happens when running DSL to generate the build configurations. That's a `MultiTestProjectBucket` so there's an assertion:

```
More than one scenario split for test project archivePerformanceProject
```

Comment (blindpirate) [reply .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:137]: Wouldn't it be two same build configuration?

Comment (blindpirate) [reply .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:137]: Ah... the scenarios in in the scenario files, I see..

Comment (wolfs) [reply .teamcity/src/main/kotlin/model/PerformanceTestBucketProvider.kt:137]: Yes, it would be the same test which we want to run multiple times. So, if I understand you correctly, the problem only happens for the flakiness split. The other splits are always fine, right?

In the flakyness split, we will/may have the test project `smallJavaMultiProject` (for example) 3 times, since we want to run all the tests from there three times - this is what flakiness detection does. So, if two of the unfiltered `smallJavaMultiProject` builds end up in the same bucket, then we would need to run the same task twice - which we can't as part of one build. So we could in theory invoke the build two times with the same task, though then we would lose the logs from the first execution.

**Comment by blindpirate** [comment]
Superseded by https://github.com/gradle/gradle/pull/17442
